test/unit/spec/PrimaryAuth_spec.js

```diff
@@ -985,15 +985,33 @@ function (Q, OktaAuth, LoginUtil, Okta, Util, AuthContainer, PrimaryAuthForm, Be
     });
 
     Expect.describe('Device Fingerprint', function () {
+      itp(`does not compute device fingerprint if security image is off, deviceFingerprinting
+        is true and useDeviceFingerprintForSecurityImage is true)`, function () {
+        spyOn(DeviceFingerprint, 'generateDeviceFingerprint').and.callFake(function () {
+          var deferred = Q.defer();
+          deferred.resolve('thisIsTheDeviceFingerprint');
+          return deferred.promise;
+        });
+        return setup({ features: { securityImage: false, deviceFingerprinting: true,
+          useDeviceFingerprintForSecurityImage: true }})
+          .then(function (test) {
+            test.setNextResponse(resSecurityImage);
+            test.form.setUsername('testuser');
+            return waitForBeaconChange(test);
+          })
+          .then(function () {
+            expect($.ajax.calls.count()).toBe(0);
+            expect(DeviceFingerprint.generateDeviceFingerprint).not.toHaveBeenCalled();
+          });
+      });
       itp(`contains fingerprint header in get security image request if deviceFingerprinting
         is true (useDeviceFingerprintForSecurityImage defaults to true)`, function () {
         spyOn(DeviceFingerprint, 'generateDeviceFingerprint').and.callFake(function () {
           var deferred = Q.defer();
           deferred.resolve('thisIsTheDeviceFingerprint');
           return deferred.promise;
         });
-        return setup({ features: { securityImage: true, deviceFingerprinting: true,
-          useDeviceFingerprintForSecurityImage: true }})
+        return setup({ features: { securityImage: true, deviceFingerprinting: true }})
```

Comment:
Is useDeviceFingerprintForSecurityImage true here?

Reply:
Yeah it is true by default.(which is what we are testing here) 

```
contains fingerprint header in get security image request if deviceFingerprinting
        is true (useDeviceFingerprintForSecurityImage defaults to true)
```